pkgs/build-support/gradle/update-deps.nix

```diff
@@ -0,0 +1,140 @@
+{ lib
+, pkgs
+, runtimeShell
+, stdenv
+, srcOnly
+, writeTextFile
+, writeShellScript
+, path
+, bubblewrap
+, coreutils
+, curl
+, jq
+, mitm-cache
+, nix
+, openssl
+, procps
+, python3
+}:
+
+let
+  getPkg = attrPath:
+    lib.getAttrFromPath
+      (lib.splitString "." (toString attrPath))
+      pkgs;
+in
+lib.makeOverridable (
+# the derivation to update
+{ pkg ? getPkg attrPath
+, pname ? null
+, attrPath ? pname
+# bwrap flags (this will be put in bash as-is)
+# this is relevant for downstream users
+, bwrapFlags ? "--ro-bind \"$PWD\" \"$PWD\""
+# deps path (relative to the package directory, or absolute)
+, depsPath ? "deps.json"
+# redirect stdout to stderr to allow the script to be used with update script combinators
+, silent ? true
+, useBwrap ? stdenv.isLinux
+}:
+
+let
+  keep = [ "MITM_CACHE_HOST" "MITM_CACHE_PORT" "MITM_CACHE_ADDRESS" "MITM_CACHE_CA" "MITM_CACHE_CERT_DIR" ];
+  gradleScript = writeShellScript "gradle-commands.sh" ''
+    set -eo pipefail
+    export http_proxy="$MITM_CACHE_ADDRESS"
+    export https_proxy="$MITM_CACHE_ADDRESS"
+    export SSL_CERT_FILE="$MITM_CACHE_CA"
+    export NIX_SSL_CERT_FILE="$MITM_CACHE_CA"
+    export GRADLE_USER_HOME="$(${coreutils}/bin/mktemp -d)"
+    export IN_GRADLE_UPDATE_DEPS=1
+    trap "${coreutils}/bin/rm -rf '$GRADLE_USER_HOME'" SIGINT SIGTERM ERR EXIT
+    cd "$(${coreutils}/bin/mktemp -d)"
+    ${coreutils}/bin/mkdir out
+    export out="$PWD/out"
+    trap "${coreutils}/bin/rm -rf '$PWD'" SIGINT SIGTERM ERR EXIT
+    source "$stdenv/setup"
+    phases="''${prePhases[*]:-} unpackPhase patchPhase ''${preConfigurePhases[*]:-} configurePhase gradleUpdateScript" genericBuild
+  '';
+  # escape for bash double quotes
+  source = srcOnly (pkg.overrideAttrs (old: {
+    mitmCache = "";
+    gradleInitScript = ./init-deps.gradle;
+  }));
+  sourceDrvPath = builtins.unsafeDiscardOutputDependency source.drvPath;
+  nixShellKeep = lib.concatMapStringsSep " " (x: "--keep ${x}") keep;
+in
+writeTextFile {
+  name = "fetch-deps.sh";
+  executable = true;
+  # see pkgs/common-updater/combinators.nix
+  derivationArgs.passthru =
+    { supportedFeatures = lib.optional silent "silent"; }
+    // lib.optionalAttrs (attrPath != null) { inherit attrPath; };
+  text = ''
+    #!${runtimeShell}
+    set -eo pipefail
+    outPath="${
+      # if this is an absolute path in nix store, use path relative to the store path
+      if lib.hasPrefix "${builtins.storeDir}/" (toString depsPath)
+      then builtins.concatStringsSep "/" (lib.drop 1 (lib.splitString "/" (lib.removePrefix "${builtins.storeDir}/" (toString depsPath))))
+      # if this is an absolute path, just use that path
+      else if lib.hasPrefix "/" (toString depsPath)
+      then toString depsPath
+      # otherwise, use a path relative to the package
+      else "${dirOf pkg.meta.position}/${depsPath}"
+    }"
+
+    pushd "$(${coreutils}/bin/mktemp -d)" >/dev/null
```

Comment:
Same here.

Reply:
done (in my local changes, will push soon)